### PR TITLE
Closes issue #96: Self healing property of SF cache

### PR DESCRIPTION
**Pull request by pritishmishra**
Description: Handling self-healing of SF cache

### CODE REVIEW

File: lib/bosh/BoshDirectorClient.js [hunk 1]
```diff
@@ -133,13 +144,13 @@ class BoshDirectorClient extends HttpClient {
         logger.silly('found director in cache...', cache_val.name);
         return cache_val;
       }
-      logger.debug('cache miss for..', deploymentName);
+      logger.debug('cache miss for..', typeof deploymentName);
```
Comment (jagadish-kb) [inline lib/bosh/BoshDirectorClient.js]: revert this log to the previous one

File: lib/bosh/BoshDirectorClient.js [hunk 2]
```diff
@@ -189,10 +215,21 @@ class BoshDirectorClient extends HttpClient {
       .then(res => JSON.parse(res.body))
       .tap(deployments => {
         this.updateCache(config, deployments);
-        logger.info('Updated cache for config - ', config.name);
+        logger.info(`Updated cache for config - ${config.name}`, deployments);
```
Comment (jagadish-kb) [inline lib/bosh/BoshDirectorClient.js]: revert to previous log, this results in a huge log

File: lib/utils/HttpClient.js [hunk 3]
```diff
@@ -164,6 +164,7 @@ class HttpClient {
           err = new InternalServerError(message);
           break;
         }
+
```
Comment (jagadish-kb) [inline lib/utils/HttpClient.js]: unnecessary line. remove it.

**Comment by pritishmishra** [comment]
Implemented review comments as part of last commit. Please check.

File: lib/bosh/BoshDirectorClient.js [hunk 4]
```diff
@@ -193,6 +217,17 @@ class BoshDirectorClient extends HttpClient {
       });
   }
 
+
```
Comment (jagadish-kb) [inline lib/bosh/BoshDirectorClient.js]: extra line

**Pull request merged**